docs/source/utils/notebook.py

```diff
@@ -1,16 +1,38 @@
+import re
+from typing import Dict, List, Optional
+
 from jupytext import jupytext
 
+start_pattern = re.compile(r'# %% \[markdown\]\n"""\n# (\d\. .*)\n\n[\S\s]*?"""\n')
```

Comment:
It seems, that will be better insert the dependencies after the first cell and just before the second, and start of the second is getting more important then content of the first cell

Reply:
Yes, it inserts dependencies right after the first cell. The second cell is usually example import definition. Do you suggest inserting requirements after that?

Reply:
Need to support several numbers in example number (can be 10, 11, 12 etc.). \d -> \d+